### ISSUE TITLE
ES6, React, Redux Support.

### ISSUE THREAD

**Issue by jpmelnik** [esdoc/esdoc#404]
IMC Class isn't recognized with **export default connect( undefined, mapDispatchToProps )( IMC );**

But, Works Fine with **export default class IMC extends Component {**...
```

/**
 * IMC
 * @extends {React Component}
 */
class IMC extends Component {
  constructor( props ) {
    super( props );
  }
  render(  ) {
    return(<div></div>);
  }
}
const mapDispatchToProps = ( dispatch ) => {
  return bindActionCreators( { calculateIMC: calculateIMC }, dispatch );
};

export default connect( undefined, mapDispatchToProps )( IMC );
```

**Comment by pronebird** [comment]
`{React Component}` = syntax error

**Comment by kevineaton** [comment]
I have the same issue. None of my classes that use Redux get processed, maybe due to the export being wrapped in the connect. This makes ESDoc something I can't really use, even though I would love to. Any way to get the class to render its documentation even though the export is at the bottom of the file?

**Comment by bengotow** [comment]
Yeah - the basic problem here (if I understand correctly) is that decorator functions like `connect` that wrap the class return a new value, which doesn't have docs, so as far as esdoc is concerned IMC is not exported from the file. 

I'd definitely be a fan of some sort of annotation, either on the class or the connect method, that indicate that it should be documented. Some sort of explicit `@exported` annotation would be really nice for node.js support too I think.

**Comment by chrisharman** [comment]
Came across what I believe to be the same issue, except that my components were being exported wrapped with `withRouter()` like so:

```
class Dashboard extends Component {
  ...
}

export default withRouter(Dashboard);
```

I changed the component to export without the wrapper:

```
export default class Dashboard extends Component {
  ...
}
```

And created a go-between file to be able to use the components like I need them (wrapped with `withRouter()`):

```
// RouterComponents.js
import { withRouter } from 'react-router-dom';
import Dashboard from './app/Dashboard';

const RouterComponents = {
  'Dashboard': withRouter(Dashboard),
  ...
};

export default RouterComponents;
```

After importing RouterComponents, I can then render the component with `<RouterComponents.Dashboard />`. This would be exceedingly obnoxious if every component was wrapped with a decorator function. For my use case, there were only a few components requiring this approach and they were already grouped together logically, so it seemed fine. ESDoc parses the Dashboard component docs properly and everything seems to work okay with this workaround, though I haven't tested the approach specifically with Redux functions.
